datatypes/src/primitives/feature_data.rs

```diff
@@ -53,25 +53,101 @@ pub enum FeatureDataRef<'f> {
     NullableCategorical(NullableCategoricalDataRef<'f>),
 }
 
+impl<'f> FeatureDataRef<'f> {
+    pub fn json_values(&self) -> Box<dyn Iterator<Item = serde_json::Value> + '_> {
```

Comment:
a way to replace the Box<dyn ... would be nice but i think this is fine for the moment.

Reply:
I can't think of an easy way to not use a vtable right now. If you have a proposal, I will try to realize it.